ui.frontend/src/utils.js

```diff
@@ -385,6 +390,78 @@ class Utils {
         }
     }
 
+    /**
+     * Registers custom functions from form json.
+     * @param {object} formJson - The Sling model exporter representation of the form
+     */
+    static async registerCustomFunctionsV3(formJson) {
+        // Extract custom function names from form events
+        const extractFunctionNames = (formJson) => {
+            const functionNames = new Set();
+            
+            // Helper function to extract function names from event strings
+            const extractFromEventString = (eventString) => {
+                // Match patterns like functionName() with optional parameters
+                const functionPattern = /(\w+)\s*\(/g;
```

Comment:
This is problematic and there are edge cases where the field name contain `(`. Not possible through UI, but with multiple ways of creating a form, it may be possible.

I would parse the event string using JSON Formula and then extract out the functions from AST.

Reply:
As discussed, we are also checking for function definition to be present in the window object before registering a custom functions which should take care of edge case that some form field might have name like abc() and it should not be registered as custom function in this case.
Also, using JSON formula for parsing and extracting function name might have performance impact on the form rendering as well. So for now we are good with this approach.